ERA/ERAv4.nlogo

```diff
@@ -110,107 +143,225 @@ end
+        __set-line-thickness 0.15
+        stamp
+        die
+      ]
+  ]
+  ask patches with [pxcor = 16] [
+    sprout 1 [
+        set color 0
+        set heading 0
+        set xcor pxcor + 0.4
+        set shape "line"
+        __set-line-thickness 0.15
+        stamp
+        die
+      ]
   ]
-  ask patches with [ecoregion = "C"][
+  ask patches with [pxcor = -16] [
     sprout 1 [
-      set color 3
-      set shape "dot"
-      stamp
-      die
+        set color 0
+        set heading 0
+        set xcor pxcor - 0.4
+        set shape "line"
+        __set-line-thickness 0.15
+        stamp
+        die
+      ]
+  ]
+
+end
+
+to setup-ERiE
+  let soil-health-table table:make
+  let erie-table table:make
+  let curr-ecoregion 0
+  while [curr-ecoregion < num-ecoregions] [
+    let eco (index-to-ecoregion curr-ecoregion)
+    ifelse count patches with [ecoregion = eco] = 0 [
+      table:put soil-health-table eco 0
+    ]  [
+      table:put soil-health-table eco mean [soil-health] of patches with [ecoregion = eco]
     ]
+    table:put erie-table eco 1
+    set curr-ecoregion curr-ecoregion + 1
   ]
+  set PREVIOUS-SOIL-HEALTH soil-health-table
+  set ERiE erie-table
 end
 
-to setup-ERiC
-  let eric-table table:make
-  table:put eric-table "ERiC 0" 1.2
-  table:put eric-table "ERiC 1" 0.8
-  table:put eric-table "ERiC 2" 1.5
-  table:put eric-table "ERiC other" 1
-  set ERiC eric-table
+to setup-international-COT  ;;currently unused but can be implemented later (would only be useful if we have a feedback loop where economic behavior is attracted by devaluing the currency (backwords intuition)
+  let cot-table table:make
+  let curr-jurisdiction 0
+  while [curr-jurisdiction < num-jurisdictions + num-decentralized-currencies] [
+    table:put cot-table curr-jurisdiction random-normal 1000 100 ;arbitrary distribution
+    set curr-jurisdiction curr-jurisdiction + 1
+  ]
+  if (calibration = "2-dominant") [
+    table:put cot-table 0 random-normal 4000 400
+    table:put cot-table 1 random-normal 4000 400
+  ]
+  set INTERNATIONAL-COT cot-table
 end
 
 to setup-base-yr-exch-rate
-  let exch-rate-0 table:make
-  table:put exch-rate-0 "0" 1
-  table:put exch-rate-0 "1" 3
-  table:put exch-rate-0 "2" 0.8
-  table:put exch-rate-0 "other" 1.2
-
-  let exch-rate-1 table:make
-  table:put exch-rate-1 "0" 0.67
-  table:put exch-rate-1 "1" 1
-  table:put exch-rate-1 "2" 2
-  table:put exch-rate-1 "other" 1.8
-
-  let exch-rate-2 table:make
-  table:put exch-rate-2 "0" 1.25
-  table:put exch-rate-2 "1" 0.5
-  table:put exch-rate-2 "2" 1
-  table:put exch-rate-2 "other" 1.6
-
-  let exch-rate-other table:make
-  table:put exch-rate-other "0" 0.83
-  table:put exch-rate-other "1" 0.56
-  table:put exch-rate-other "2" 0.625
-  table:put exch-rate-other "other" 1
-
-  set BASE-YEAR-EXCHANGE-RATES (list exch-rate-0 exch-rate-1 exch-rate-2 exch-rate-other)
+
+  let base-exchange-rates (list)
+  let currency-row 0
+  let exch-rate table:make
```

Comment:
I think this is useful from a data organization standpoint, but in terms of implementation, I am a little confused how we would do this. I thought of two possible implementations. 1. We could do undirected links, and assume the "base currency" for the exchange rate is the currency with the lower number. 2. Each pair of nodes has two directional links (one for each direction) where the value reports the exchange rate. I would lean more towards the second implementation because it doesn't use an arbitrary numbering system.